File: BalancedArray.java
import java.util.Arrays;
import java.util.Scanner;

public class BalancedArray {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        int i = 0;
        while (i < t) {
            int n = sc.nextInt();
            if ((n / 2) % 2 != 0) {
                System.out.println("NO");
            } else {
                int[] arr = new int[n];
                int k = 2;
                int p = 1;
                int fsum = 0;
                int lsum = 0;

                for (int j = 0; j < n / 2; j++) {
                    arr[j] = k;
                    fsum += k;
                    k += 2;
                }
                for (int l = n / 2; l < n - 1; l++) {
                    arr[l] = p;
                    lsum += p;
                    p += 2;
                }

                int pLsum=lsum;
                while(fsum!=lsum){
                    if(fsum<lsum+(p+2)){
                        break;
                    }
                    p+=2;
                    lsum+=p;
                }
                if(fsum == pLsum+p){
                    arr[n-1]=p;
                    System.out.println("YES");
                    for(int v=0;v<arr.length;v++){
                        System.out.printf("%d ",arr[v]);
                    }
                    System.out.println();
                }
                else{
                    System.out.println("NO");
                }
            }
            i++;
        }
        sc.close();
    }
}

File: codeforcebyme/codeforce/BalancedArray.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int t;
    cin >> t;
    for (int i = 0; i < t; ++i) {
        int n;
        cin >> n;
        if ((n / 2) % 2 != 0) {
            cout << "NO" << endl;
        } else {
            vector<int> arr(n);
            int k = 1;
            int fsum = 0, lsum = 0;
            for (int j = 0; j < n - 1; ++j) {
                if (j < n / 2) {
                    arr[j] = 2 * j + 2;
                    fsum += arr[j];
                } else {
                    arr[j] = k;
                    lsum += arr[j];
                    k += 2;
                }
            }
            while (lsum + k != fsum) {
                if (lsum + k > fsum) {
                    break;
                }
                k += 2;
            }
            if (fsum == lsum + k) {
                arr[n - 1] = k;
                cout << "YES" << endl;
                for (int q : arr) {
                    cout << q << " ";
                }
                cout << endl;
            } else {
                cout << "NO" << endl;
            }
        }
    }
    return 0;
}
